shell/browser/native_window_mac.mm

```diff
@@ -875,52 +875,62 @@ void ViewDidMoveToSuperview(NSView* self, SEL _cmd) {
 }
 
 void NativeWindowMac::SetAlwaysOnTop(ui::ZOrderLevel z_order,
-                                     const std::string& level,
-                                     int relativeLevel,
-                                     std::string* error) {
-  int windowLevel = NSNormalWindowLevel;
-  bool level_changed = z_order != widget()->GetZOrderLevel();
-  CGWindowLevel maxWindowLevel = CGWindowLevelForKey(kCGMaximumWindowLevelKey);
-  CGWindowLevel minWindowLevel = CGWindowLevelForKey(kCGMinimumWindowLevelKey);
-
-  if (z_order != ui::ZOrderLevel::kNormal) {
-    if (level == "floating") {
-      windowLevel = NSFloatingWindowLevel;
-    } else if (level == "torn-off-menu") {
-      windowLevel = NSTornOffMenuWindowLevel;
-    } else if (level == "modal-panel") {
-      windowLevel = NSModalPanelWindowLevel;
-    } else if (level == "main-menu") {
-      windowLevel = NSMainMenuWindowLevel;
-    } else if (level == "status") {
-      windowLevel = NSStatusWindowLevel;
-    } else if (level == "pop-up-menu") {
-      windowLevel = NSPopUpMenuWindowLevel;
-    } else if (level == "screen-saver") {
-      windowLevel = NSScreenSaverWindowLevel;
-    } else if (level == "dock") {
-      // Deprecated by macOS, but kept for backwards compatibility
-      windowLevel = NSDockWindowLevel;
-    }
+                                     const std::string& level_name,
+                                     int relative_level) {
+  if (z_order == ui::ZOrderLevel::kNormal) {
+    SetWindowLevel(NSNormalWindowLevel);
+    return;
   }
 
-  NSInteger newLevel = windowLevel + relativeLevel;
-  if (newLevel >= minWindowLevel && newLevel <= maxWindowLevel) {
-    was_maximizable_ = IsMaximizable();
-    [window_ setLevel:newLevel];
-    // Set level will make the zoom button revert to default, probably
-    // a bug of Cocoa or macOS.
-    [[window_ standardWindowButton:NSWindowZoomButton]
-        setEnabled:was_maximizable_];
-  } else {
-    *error = std::string([[NSString
-        stringWithFormat:@"relativeLevel must be between %d and %d",
```

Comment:
I removed this error in favor of a clamp because it's an absurdly wide range (-2147483643, 2147483631) and I didn't see much value in the error. Willing to add back on strong feelings, though.